verify-loader

```diff
@@ -135,12 +152,25 @@ def verify(input_gen, report_interval=REPORT_INTERVAL):
     try:
         for line in input_gen:
             line_len = len(line)
-            if not line.startswith("loader seq - "):
+            line = get_loader_line(line, matchsubstring)
+            if not line:
                 report_ignored_bytes += line_len
                 report_ignored_count += 1
             else:
                 try:
-                    _, invocid, seqval, payload = line.split('-', 4)
+                    if matchsubstring:
```

Comment:
@richm, it looks like `get_loader_line` already returns the original in both cases, you do we even need lines 161 - 172?

Reply:
@portante because of the extra stuff at the end of the line, in the case that matchsubstring is handling

Reply:
@portante that is, do we want to measure the exact byte count of the original payload, or the payload byte count + the extra stuff/metadata that comes after the payload?

Reply:
Not sure I follow: do you mean to say that if the loader emits: "AAAAA" and after that message flows through a logging pipeline it becomes "aaaaaAAAAAzzzzz" or "aaaaaAAAAA", we want to count the bytes only for "AAAAA"?

It looks to me like the code we have will take messages "aaaaaAAAAAzzzzz" and "aaaaaAAAAA" and turn them into "AAAAAzzzzz" and "AAAAA", respectively.

I don't think we embed the size of the string in the loader message, though we could do that and then be able to measure both.

Or maybe I don't understand the situation well.

Reply:
When the `$CUSTOMER` logging pipeline emits a record it read from the loader input, it will look like "xxxxxx loader seq - loader counter - loader payload xxxxxxxxxxxxxxxxxxxxxxxxxxxxBunchOfOtherStuff"

In the old code that just dealt with "loader seq - loader counter - loader payload" it measured the number of bytes read and reported as 
```
                        report_bytes += line_len
```
In that case, the line_len == loader payload size

In the new case, line_len == loader payload size + a lot of other stuff

When we calculate the report_bytes, for the customer pipeline case, should it be the sum of len("loader seq - loader counter - loader payload") or the sum of len("xxxxxx loader seq - loader counter - loader payload xxxxxxxxxxxxxxxxxxxxxxxxxxxxBunchOfOtherStuff")
?

Reply:
or do we need to have two measurements in this case - total_message_size, which is the sum of len("loader seq - loader counter - loader payload") or the sum of len("xxxxxx loader seq - loader counter - loader payload xxxxxxxxxxxxxxxxxxxxxxxxxxxxBunchOfOtherStuff"), and report_bytes which is just sum of len("loader seq - loader counter - loader payload")?  or vice versa

Reply:
I'm recreating my test environment, then I can show you exactly what the messages look like.  Will take a while

Reply:
@portante here you go
```
<14>1 Apr 13 22:11:09 kubernetes.default.svc ocp.rsyslog_test.container: { "kubernetes": { "namespace_id": "99f4cd21-3f64-11e8-9cbf-545200666b2b", "creation_timestamp": "2018-04-13T21:49:54Z", "pod_id": "83d91828-3f67-11e8-857d-545200666b2b", "labels": { "loader": "true", "application": "", "deploymentConfig": "", "microservice": "" }, "host": "oshift-ocp-37.oshift-ocp-37.test", "pod_name": "loader-1", "namespace_name": "logging", "container_name": "", "master_url": "https:\/\/kubernetes.default.svc:443" }, "docker": { "container_id": "deb50dc28d2f762431826fa6ea991f745c3d15fe6110b8474c2ebd75864fa24b" }, "hostname": "oshift-ocp-37.oshift-ocp-37.test", "message": "loader seq - f1a37ee3d9244be8bc0909d0fcc0bd57 - 0000035678 - fefy8gz47nvlgb6xn71s08otjitz02fp0i8akfcsm7q1qfyfrnikg7lzg77xit5skme5tmbnjtkewjt3ue4h0zdwrjamqvep435e1hw67tio1f6khvmq0zpe72sye0lkrpqrz61zm0u35qfdaz9wfyv071f4tfkvri835lwd63fkpc6645l1ata9uecti8aezmbnzqogh5f2eiadgoh2ln2wz45ki7z58tl21eld9rbowq3ebxapuklp63j11o7s92ffc2ijruj6lpall8q1bxahaebx55st3kvqywbluyct2nrrc3s5zynwlwrrti9fe7a8c282iybjtyyyjev7dmti8ijsyxms76p65wqir6h40cmfkbbm41y6bpfjbexk116xcjmkziheg4qq7a2whpwkq5b2kgvy5s72hraxolony7o0kuokbo08nlzdgfdgm0se3ul5nggatbazpj5yvpj8x7stdjqqbgsolugarj0xi3btf9x23wzkq3mxcbhnavmxylpp02hkr4tk3qdzo3mv7gzz36bw0hvx3d96zrkr6d65jgw2o4c5iskzkc2wcjhtv5thq5fqsjs2a7607n6jkdo2j61glcaay4vva4qvjnitn20c3o90kemge06zd1delce0pmj8100e6regp8gbgi69vflux6au265pqxy7vjbs1ndsp2dcpsx1119zeo081km5dryavcybe9hslk1hak1fd73ncyocoun6u7dd452zloesln22skm8djmcizaegndzx06ag9vd7ksz1r2aj3aom84uyxudkyb8lgp7bsqhajv1sdvidw2wx4j0yr13t5pdx1p70y1lyx5qc2ho4yf4rjfv90ut45nk28nww5da7hcncgos9ai7zebwzqx64mnhef2h507d67a3rztd9g57qb8r5ewjfemksv3n1o9j7fplieqetnpjklqd73qz1suotu1lkp4bd0mh8mgtnp3ykb5z0o7le", "@timestamp": "2018-04-13T22:11:09.578078+00:00" }
```

Reply:
It will add a lot of overhead to verify-loader to parse json - I thought about doing that initially, but decided just simple string parsing would be faster - but perhaps json parsing is just as expensive as regex matching